MerlinAU.sh

```diff
@@ -1631,74 +1631,89 @@ if false ; then echo "388.5.0" ; return 0 ; fi
    return 1
 }
 
-##-------------------------------------##
-## Added by Martinski W. [2023-Dec-15] ##
-##-------------------------------------##
+##----------------------------------------##
+## Modified by Martinski W. [2024-Apr-01] ##
+##----------------------------------------##
 #---------------------------------------------------------------------#
 # The actual amount of RAM that is available for any new process
 # (*without* using the swap file) can be roughly estimated from
-# "MemFree" & "Page Cache" (i.e. Active files + Inactive files),
+# "Free Memory" & "Page Cache" (e.g. Inactive memory pages).
 # This estimate must take into account that the overall system
 # (kernel + native services + tmpfs) needs a minimum amount of RAM
 # to continue to work, and that not all reclaimable Page Cache can
 # be reclaimed because some may actually be in used at the time.
+# NOTE: [Martinski]
+# Since reported "Available RAM" estimates tend to be extremely 
+# conservative in many cases, we decided to take another approach
+# and calculate it based on the reported "Free RAM" plus ~66% of 
+# reported "Memory Cached" and then take the largest of the two
+# values: Reported "Available RAM" vs Calculated "Available RAM"
+# While still somewhat conservative, this would provide a better
+# estimate, especially at the time when the router is about to
+# shut down and terminate all non-critical services/processes
+# before the actual F/W flash is performed.
 #---------------------------------------------------------------------#
 _GetAvailableRAM_KB_()
 {
-   local theMemAvailable_KB  theMemFree_KB
-   local activeFiles_KB  inactiveFiles_KB  thePageCache_KB
-
-   theMemAvailable_KB="$(cat /proc/meminfo | awk -F ' ' '/^MemAvailable: /{print $2}')"
-   [ -n "$theMemAvailable_KB" ] && echo "$theMemAvailable_KB" && return 0
-
-   theMemFree_KB="$(cat /proc/meminfo | awk -F ' ' '/^MemFree: /{print $2}')"
-   activeFiles_KB="$(cat /proc/meminfo | awk -F ' ' '/^Active\(file\): /{print $2}')"
-   inactiveFiles_KB="$(cat /proc/meminfo | awk -F ' ' '/^Inactive\(file\): /{print $2}')"
-   thePageCache_KB="$((activeFiles_KB + inactiveFiles_KB))"
-
-   #----------------------------------------------------------------#
-   # Since not all Page Cache is guaranteed to be reclaimed at any
-   # moment, we simply estimate that only half will be reclaimable.
-   #----------------------------------------------------------------#
-   theMemAvailable_KB="$((theMemFree_KB + (thePageCache_KB / 2)))"
+   local theMemAvailable_KB  theMemAvail_KB  theMemCache_KB
+   local theMemFree1_KB  theMemFree2_KB  inactivePgs_KB
+
+   _MaxNumber_() { echo "$(($1 < $2 ? $2 : $1))" ; }
+
+   theMemCache_KB="$(awk -F ' ' '/^Cached:/{print $2}' /proc/meminfo)"
+   theMemFree1_KB="$(awk -F ' ' '/^MemFree:/{print $2}' /proc/meminfo)"
+   theMemAvail_KB="$(awk -F ' ' '/^MemAvailable:/{print $2}' /proc/meminfo)"
+   # Assumes that only ~66% of Page Cache can be reclaimed #
```

Comment:
I prefer the higher accuracy when comparing to conservatism.
Nice work, I like the fallback for the theMemAvailable_KB variable if MemAvailable is not available.